benchmarks/scripts/plot_avg_benchmarks_over_time.py

```diff
@@ -62,17 +60,27 @@ def extract_compiler_versions(header):
 color_map = {
     compiler: colormap(i) for i, compiler in enumerate(unique_compilers)
 }
+unique_versions = sorted(df_dates["compiler_version"].unique())
+dates = []
+for version in unique_versions:
+    df_versions = df_dates[df_dates["compiler_version"] == version]
```

Comment:
Is this `compiler_version` only the semver string? If so, is it possible that this might mix the same version of different packages? For example `cirq 1.2.0` and `qiskit 1.2.0`? If so, I think that might end up taking the earliest date across the union of those two.